nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/TransformJSON.java

```diff
@@ -0,0 +1,212 @@
+            .addValidator(new JOLTSpecValidator())
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor JOLT_TRANSFORM = new PropertyDescriptor.Builder()
+            .name("Jolt Transformation")
+            .description("Specifies the Jolt Transformation that should be used with the provided specification.")
+            .required(true)
+            .allowableValues(SHIFTR, CHAINR, DEFAULTR, REMOVR)
+            .defaultValue(CHAINR.getValue())
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("The FlowFile with transformed content will be routed to this relationship")
+            .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("If a FlowFile fails processing for any reason (for example, the FlowFile is not valid JSON), it will be routed to this relationship")
+            .build();
+
+    private List<PropertyDescriptor> properties;
+    private Set<Relationship> relationships;
+    private Transform transform;
+
+
+    @Override
+    protected void init(ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(JOLT_TRANSFORM);
+        properties.add(JOLT_SPEC);
+        this.properties = Collections.unmodifiableList(properties);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, ProcessSession session) throws ProcessException {
+
+        final FlowFile original = session.get();
+        if (original == null) {
+            return;
+        }
+
+        final ProcessorLog logger = getLogger();
+        final StopWatch stopWatch = new StopWatch(true);
+
+        try {
+
+            FlowFile transformed = session.write(original, new StreamCallback() {
+                @Override
+                public void process(final InputStream rawIn, final OutputStream out) throws IOException {
+
+                    try (final InputStream in = new BufferedInputStream(rawIn)) {
+                        Object inputJson = JsonUtils.jsonToObject(in);
+                        Object transformedJson = transform.transform(inputJson);
+                        out.write(JsonUtils.toJsonString(transformedJson).getBytes());
+                    } catch (final Exception e) {
+                        throw new IOException(e);
```

Comment:
Hi Oleg, based on your comments I used IntelliJ's coverage tools and confirmed this section of code in question was reached.  The test TestTransformJSON.testInvalidFlowFileContent triggers the exception. If the flow file content isn't json the utility class I'm using to parse (which is backed by Jackson) internally catches a JsonParseException and rethrows it as a RuntimeException. If it allowed the JsonParseException through it would have been fine, since it extends the IOException. Not quite sure why they wrapped it in a RuntimeException. 
Anyway I did confirm that not having the catch block misses the ProcessException altogether and my test fails, since it's checking that flow files were routed to the failure relationship.
Also I did add a couple of tests to ensure 100% coverage.

Will commit and push changes shortly for review.